app/backend/src/tests/test_threads.py

```diff
@@ -0,0 +1,125 @@
+import contextlib
+import string
+
+import grpc
+import pytest
+from google.protobuf import empty_pb2, wrappers_pb2
+from sqlalchemy.sql import and_
+
+from couchers import errors
+from couchers.db import session_scope
+from couchers.models import Thread
+from couchers.servicers.threads import Threads
+from couchers.utils import now
+from pb import threads_pb2, threads_pb2_grpc
+from tests.test_fixtures import api_session, db, fake_channel, generate_user, testconfig
+
+
+@pytest.fixture(autouse=True)
+def _(testconfig):
+    pass
+
+
+@contextlib.contextmanager
+def threads_session(token):
+    """
+    Create a communities API for testing, uses the token for auth
+    """
+    channel = fake_channel(token)
+    threads_pb2_grpc.add_ThreadsServicer_to_server(Threads(), channel)
+    yield threads_pb2_grpc.ThreadsStub(channel)
+
+
+def test_threads_basic(db):
+    user1, token1 = generate_user()
+    user2, token2 = generate_user()
+
+    PARENT_THREAD_ID = 10
+
+    # Create a dummy Thread (should be replaced by pages later on)
+    with session_scope() as session:
+        session.add(Thread(id=1, title="foo"))
+
```

Comment:
How about this?

```py
thread = Thread(id=1, title="foo")
session.add(thread)
session.flush()
PARENT_THREAD_ID = thread.id
```

I try to shy away from implementation details and execution details in tests and rely only on the public API. (i.e. the thread number is impl detail, and that it ends up having ID 10 is also a detail of execution)

Reply:
Thanks for the tip - applied